enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java

```diff
@@ -159,4 +148,114 @@ void testConvertUiRiskFlagsXCF() throws Exception {
+                .isNotNull()
+                .extracting(Operation::getUi)
+                .isNotNull()
+                .extracting(UiExtensions::getPostApprovalScreen)
+                .isNotNull()
+                .returns("Thank you for your order", from(PostApprovalScreen::getHeading))
+                .returns("You will be redirected to the merchant application.", from(PostApprovalScreen::getMessage))
+                .extracting(PostApprovalScreen::getPayload)
+                .isInstanceOf(PostApprovalScreen.MerchantRedirectPayload.class)
+                .isEqualTo(expectedPayload);
+    }
+
+    @Test
+    void testConvertUiPostApprovalMerchantRedirectWithSubstitutedUrl() throws Exception {
+        final OperationDetailResponse operationDetail = createOperationDetailResponse();
+        operationDetail.setParameters(Map.of("userId", "666", "redirectUrl", "https://www.example.com"));
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+        operationTemplate.setUi("{\n" +
+                "  \"postApprovalScreen\": {\n" +
+                "    \"heading\": \"Thank you for your order\",\n" +
+                "    \"message\": \"You will be redirected to the merchant application.\",\n" +
+                "    \"payload\": {\n" +
+                "      \"type\": \"MERCHANT_REDIRECT\",\n" +
+                "      \"redirectText\": \"Go to the application\",\n" +
+                "      \"redirectUrl\": \"${redirectUrl}\",\n" +
+                "      \"countdown\": 5\n" +
+                "    }\n" +
+                "  }\n" +
+                "}");
+
+        final Operation result = tested.convert(operationDetail, operationTemplate);
+
+        final var expectedPayload = new PostApprovalScreen.MerchantRedirectPayload();
+        expectedPayload.setRedirectText("Go to the application");
+        expectedPayload.setRedirectUrl("https://www.example.com");
+        expectedPayload.setCountdown(5);
+
+        assertThat(result)
+                .isNotNull()
+                .extracting(Operation::getUi)
+                .isNotNull()
+                .extracting(UiExtensions::getPostApprovalScreen)
+                .isNotNull()
+                .returns("Thank you for your order", from(PostApprovalScreen::getHeading))
+                .returns("You will be redirected to the merchant application.", from(PostApprovalScreen::getMessage))
+                .extracting(PostApprovalScreen::getPayload)
+                .isInstanceOf(PostApprovalScreen.MerchantRedirectPayload.class)
+                .isEqualTo(expectedPayload);
+    }
+
+    @Test
+    void testConvertUiPostApprovalInfoMessage() throws Exception {
+        final OperationDetailResponse operationDetail = createOperationDetailResponse();
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+        operationTemplate.setUi("{\n" +
+                "  \"postApprovalScreen\": {\n" +
```

Comment:
This is an example of generic _post approval screen_. Payload is missing on purpose. We may cover such scenario without discriminator.

Reply:
My understanding is that the payload for the `GENERIC` subtype should not be missing, but it should be customizable using a subtitutor. This way the backend can support generic functionality which is added to the app before next backend release, using parameter substitution. @kober32 do we agree on this use case?